src/features/payment/PaymentItem.tsx

```diff
@@ -187,10 +185,8 @@ const PaymentItem = ({
   }, [index, onRemove])
 
   const isDeepLink = useMemo(
-    () =>
-      address &&
-      !(l1Network === 'helium' ? account?.address : account?.solanaAddress),
-    [account, address, l1Network],
+    () => address && addressIsWrongNetType,
```

Comment:
Hmm we won't need to check the `netType` on Solana. I suggest we check if the `l1Network  === helium` otherwise we don't check the `netType`

Reply:
Thanks!

Reply:
Hmm if its a deeplink then this condition needs to be met `address && !account?.address`. I think this might break typing in an address for helium L1